File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Collections
Imports DevExpress.XtraVerticalGrid.ViewInfo
Imports DevExpress.XtraVerticalGrid.Events
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.DXErrorProvider

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form


		Public Sub New()
			InitializeComponent()
		End Sub

		Private Function GetRowInfo(ByVal e As CustomDrawRowValueCellEventArgs) As BaseRowViewInfo
			Dim rowsViewInfo As ArrayList = propertyGridControl1.ViewInfo.RowsViewInfo
			For i As Integer = 0 To rowsViewInfo.Count - 1
				Dim info As BaseRowViewInfo = TryCast(rowsViewInfo(i), BaseRowViewInfo)
                If info.Row.Equals(e.Row) Then
					Return info
				End If
			Next i
			Return Nothing
		End Function

		Public Function GetEditorViewInfo(ByVal rowInfo As BaseRowViewInfo, ByVal e As CustomDrawRowValueCellEventArgs) As BaseEditViewInfo
			If rowInfo Is Nothing Then
				Return Nothing
			End If
			For i As Integer = 0 To rowInfo.ValuesInfo.Count - 1
				Dim valuesInfo As RowValueInfo = rowInfo.ValuesInfo(i)
				If valuesInfo.RecordIndex = e.RecordIndex AndAlso valuesInfo.RowCellIndex = e.CellIndex Then
					Return valuesInfo.EditorViewInfo
				End If
			Next i
			Return Nothing
		End Function

		Private Sub propertyGridControl1_CustomDrawRowValueCell(ByVal sender As Object, ByVal e As CustomDrawRowValueCellEventArgs) Handles propertyGridControl1.CustomDrawRowValueCell
			If e.CellValue IsNot Nothing Then
				Return
			End If
			Dim rowInfo As BaseRowViewInfo = GetRowInfo(e)
			Dim editViewInfo As BaseEditViewInfo = GetEditorViewInfo(rowInfo, e)
			editViewInfo.ErrorIconText = "IsNull"
			editViewInfo.ShowErrorIcon = True
			editViewInfo.FillBackground = True
			editViewInfo.ErrorIcon = DXErrorProvider.GetErrorIconInternal(ErrorType.Critical)
			editViewInfo.CalcViewInfo(e.Graphics)
		End Sub



	End Class
End Namespace

File: CS/C#/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Collections;
using DevExpress.XtraVerticalGrid.ViewInfo;
using DevExpress.XtraVerticalGrid.Events;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.DXErrorProvider;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {

       
        public Form1()
        {
            InitializeComponent();
        }

        private BaseRowViewInfo GetRowInfo(CustomDrawRowValueCellEventArgs e)
        {
            ArrayList rowsViewInfo = propertyGridControl1.ViewInfo.RowsViewInfo;
            for (int i = 0; i < rowsViewInfo.Count; i++)
            {
                BaseRowViewInfo info = rowsViewInfo[i] as BaseRowViewInfo;
                if (info.Row == e.Row)
                    return info;
            }
            return null;
        }

        public BaseEditViewInfo GetEditorViewInfo(BaseRowViewInfo rowInfo, CustomDrawRowValueCellEventArgs e)
        {
            if (rowInfo == null) return null;
            for (int i = 0; i < rowInfo.ValuesInfo.Count; i++)
            {
                RowValueInfo valuesInfo = rowInfo.ValuesInfo[i];
                if (valuesInfo.RecordIndex == e.RecordIndex && valuesInfo.RowCellIndex == e.CellIndex)
                    return valuesInfo.EditorViewInfo;
            }
            return null;
        }

        private void propertyGridControl1_CustomDrawRowValueCell(object sender, CustomDrawRowValueCellEventArgs e)
        {
            if (e.CellValue != null) return;
            BaseRowViewInfo rowInfo = GetRowInfo(e);
            BaseEditViewInfo editViewInfo = GetEditorViewInfo(rowInfo, e);
            editViewInfo.ErrorIconText = "IsNull";
            editViewInfo.ShowErrorIcon = true;
            editViewInfo.FillBackground = true;
            editViewInfo.ErrorIcon = DXErrorProvider.GetErrorIconInternal(ErrorType.Critical);
            editViewInfo.CalcViewInfo(e.Graphics);
        }



    }
}